server/public_routes.go

```diff
@@ -87,12 +89,18 @@ func PublicRoutes(app *app.App) []*route.HandledRoute {
 		)
 	}
 
-	for providerName := range app.OauthProviders {
+	for providerName, provider := range app.OauthProviders {
+		var returnRoute *route.Route
+		if provider.ReturnMethod() == http.MethodPost {
```

Comment:
For the sake of consistency, I suggest we implement both HTTP methods for all OAuth providers instead of making it conditional.

Reply:
Think I prefer keeping the surface area as small as possible here.